examples/flux_operator_ca_hpa/application_ca_hpa_metrics.py

```diff
@@ -0,0 +1,161 @@
+                pods_for_flux[name]["waiting_for_container"] = {"current_status": True, "start": None, "end": None, "duration": None}
+            else:
+                pods_for_flux[name]["waiting_for_scheduled"] = {"current_status": False, "start": datetime_utcnow_str(), "end": None, "duration": None}
+                pods_for_flux[name]["waiting_for_container"] = {"current_status": False, "start": datetime_utcnow_str(), "end": None, "duration": None}
+        
+        elif event_type == "MODIFIED":
+            for condition in status['conditions']:
+                if condition['type'] == 'PodScheduled' and condition['status'] == 'True' and (not pods_for_flux[name]["waiting_for_scheduled"]["current_status"]):
+                    pods_for_flux[name]["waiting_for_scheduled"]["current_status"] = True
+                    pods_for_flux[name]['waiting_for_scheduled']['end'] = datetime_utcnow_str()
+                    pods_for_flux[name]['waiting_for_scheduled']['duration'] = (datetime.utcnow() - str_to_datetime(pods_for_flux[name]['waiting_for_scheduled']['start'])).total_seconds()
+                elif condition['type'] == 'Initialized':
+                    pass
+                elif condition['type'] == 'Ready':
+                    pass
+                elif condition['type'] == 'ContainersReady':
+                    pass
+        
+            if "containerStatuses" in status.keys():
+                for container_status in status["containerStatuses"]:
+                    if container_status["name"] == "flux-sample" and container_status["ready"] == True and (not pods_for_flux[name]["waiting_for_container"]["current_status"]):
+                        pods_for_flux[name]["waiting_for_container"]["current_status"] = True
+                        pods_for_flux[name]["waiting_for_container"]["end"] = datetime_utcnow_str()
+                        pods_for_flux[name]["waiting_for_container"]["duration"] = (datetime.utcnow() - str_to_datetime(pods_for_flux[name]["waiting_for_container"]["start"])).total_seconds()
+        elif event_type == "DELETED":
+            
+            for container_status in status["containerStatuses"]:
+                pods_for_flux[name]["pod_started"] =  container_status['state']['terminated']["startedAt"]
+                pods_for_flux[name]["pod_finished"] =  container_status['state']['terminated']["finishedAt"]
+                pods_for_flux[name]["pod_container_duration"] = (str_to_datetime(pods_for_flux[name]["pod_finished"]) - str_to_datetime(pods_for_flux[name]["pod_started"])).total_seconds()
+        
+        utils.write_json(pods_for_flux, output_dir)
+        print(f"🫛 POD Event")
+
+def watch_cluster_autoscaler(k8s_client, output_dir, namespace='kube-system'):
+
+    cluster_autoscaler_pod_name = None
+    response = k8s_client.list_namespaced_pod(namespace=namespace)
+
+    for pod in response.items:
+        
+        if 'cluster-autoscaler' in pod.metadata.name:
```

Comment:
Do we assume there is only one (or do we need / want a break?) And do you want to check after this loop that the name still isn't None?